src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -278,6 +284,44 @@ var ReactCompositeComponentMixin = {
     this._pendingReplaceState = false;
     this._pendingForceUpdate = false;
 
+    var markup;
+    if (inst.handleError) {
+      var checkpoint = transaction.checkpoint();
+      try {
+        markup = this.performInitialMount(renderedElement, nativeParent, nativeContainerInfo, transaction, context);
```

Comment:
Yeah, we should probably do that – though I don't like making the stack deeper unnecessarily because it hampers debugging, and performance in the case that an exception *is* thrown is probably not a huge concern. Would be nice if the common path was broken up intelligently though.